chainer/testing/link.py

```diff
@@ -0,0 +1,549 @@
+        return params_inits
+
+    def _create_link(self, initializers, backend_config):
+        link = self.create_link(initializers)
+        if not isinstance(link, chainer.Link):
+            raise TypeError(
+                '`create_link` must return a chainer.Link object.')
+
+        link.to_device(backend_config.device)
+
+        return link
+
+    def _create_initialized_link(
+            self, inits, backend_config, return_inputs_outputs=False):
+        inits = [_get_initializer(i) for i in inits]
+        link = self._create_link(inits, backend_config)
+
+        # Generate inputs and compute a forward pass to initialize the
+        # parameters.
+        inputs_np = self._generate_inputs()
+        inputs_xp = backend_config.get_array(inputs_np)
+        inputs_xp = self._to_noncontiguous_as_needed(inputs_xp)
+        input_vars = [chainer.Variable(i) for i in inputs_xp]
+        output_vars = self._forward(link, input_vars, backend_config)
+
+        link.cleargrads()
+
+        ret = link
+
+        if return_inputs_outputs:
+            outputs_xp = [v.array for v in output_vars]
+            ret = ret, inputs_xp, outputs_xp
+
+        return ret
+
+    def _generate_inputs(self):
+        inputs = self.generate_inputs()
+
+        test._check_array_types(inputs, backend.CpuDevice(), 'generate_inputs')
+
+        return inputs
+
+    def _forward(self, link, inputs, backend_config):
+        assert all(isinstance(x, chainer.Variable) for x in inputs)
+
+        with backend_config:
+            outputs = self.forward(link, inputs, backend_config.device)
+        test._check_variable_types(
+            outputs, backend_config.device, 'forward', LinkTestError)
+
+        return outputs
+
+    def _forward_expected(self, inputs, params):
+        assert all(isinstance(x, numpy.ndarray) for x in inputs)
+        assert all(isinstance(x, numpy.ndarray) for x in params)
+
+        outputs = self.forward_expected(inputs, params)
+        test._check_array_types(inputs, backend.CpuDevice(), 'test_forward')
+
+        return outputs
+
+    def _generate_grad_outputs(self, outputs_template):
+        assert all(isinstance(x, numpy.ndarray) for x in outputs_template)
+
+        grad_outputs = self.generate_grad_outputs(outputs_template)
+        test._check_array_types(
+            grad_outputs, backend.CpuDevice(), 'generate_grad_outputs')
+
+        return grad_outputs
+
+    def _to_noncontiguous_as_needed(self, contig_arrays):
+        if self.contiguous is None:
+            # non-contiguous
+            return array_module._as_noncontiguous_array(contig_arrays)
+        if self.contiguous == 'C':
+            # C-contiguous
+            return contig_arrays
+        assert False, (
+            'Invalid value of `contiguous`: {}'.format(self.contiguous))
+
+
+def _check_generated_initializer(init):
+    if isinstance(init, InitializerPair):
+        init = init.second
+        if init is None:
```

Comment:
Why comparing with `None`?
How about calling `_check_is_initializer_like`?

Reply:
`None` actually passes `_check_is_initializer_like`, maybe it shouldn't.

Reply:
Fixed to not pass None.